category/basic.lean

```diff
@@ -15,6 +17,85 @@ variables {f : Type u → Type v} [applicative f]
 lemma pure_seq_eq_map : ∀ {α β : Type u} (g : α → β) (x : f α), pure g <*> x = g <$> x :=
 @applicative.pure_seq_eq_map f _
 
+lemma pure_seq_pure {α β : Type u} (g : α → β) (x : α) :
+  pure g <*> (pure x : f α) = pure (g x) :=
+by simp [pure_seq_eq_map,applicative.map_pure]
+
+open function
+
+lemma map_seq {β γ σ : Type u} (h : γ → σ) (x : f (β → γ)) (y : f β) :
+  h <$> (x <*> y) = (comp h <$> x) <*> y :=
+by rw [← pure_seq_eq_map,← pure_seq_eq_map
+     ,applicative.seq_assoc
```

Comment:
Lean style is to put the comma at the end of the previous line.
Also, there is a space after comma.

Reply:
When the list spans multiple lines, do you align with the inside of `[` like this:

```
by rw [← applicative.pure_seq_eq_map,
       seq_assoc, map_pure,
       applicative.pure_seq_eq_map]
```

Reply:
Good question. I think in this case I would also align it with the inside of `[`.